backend/src/schema/resolvers/users/location.js

```diff
@@ -110,44 +105,36 @@ const createOrUpdateLocations = async (userId, locationName, driver) => {
   if (data.context) {
     await asyncForEach(data.context, async ctx => {
       await createLocation(session, ctx)
-      try {
-        await session.writeTransaction(transaction => {
-          return transaction.run(
-            `
+      await session.writeTransaction(transaction => {
+        return transaction.run(
+          `
               MATCH (parent:Location {id: $parentId}), (child:Location {id: $childId})
               MERGE (child)<-[:IS_IN]-(parent)
               RETURN child.id, parent.id
             `,
-            {
-              parentId: parent.id,
-              childId: ctx.id,
-            },
-          )
-        })
-        parent = ctx
-      } finally {
-        session.close()
-      }
+          {
+            parentId: parent.id,
+            childId: ctx.id,
+          },
+        )
+      })
+      parent = ctx
     })
   }
   // delete all current locations from user and add new location
-  try {
-    await session.writeTransaction(transaction => {
-      return transaction.run(
-        `
+  await session.writeTransaction(transaction => {
+    return transaction.run(
+      `
           MATCH (user:User {id: $userId})-[relationship:IS_IN]->(location:Location)
           DETACH DELETE relationship
           WITH user
-          MATCH (location:Location {id: $locationId}) 
-          MERGE (user)-[:IS_IN]->(location) 
+          MATCH (location:Location {id: $locationId})
+          MERGE (user)-[:IS_IN]->(location)
           RETURN location.id, user.id
         `,
-        { userId: userId, locationId: data.id },
-      )
-    })
-  } finally {
-    session.close()
-  }
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-19T14:33:51Z" title="Thursday, December 19th 2019, 3:33:51 pm +01:00">Dec 19, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -110,44 +105,36 @@ const createOrUpdateLocations = async (userId, locationName, driver) => {
   if (data.context) {
     await asyncForEach(data.context, async ctx => {
       await createLocation(session, ctx)
-      try {
-        await session.writeTransaction(transaction => {
-          return transaction.run(
-            `
+      await session.writeTransaction(transaction => {
+        return transaction.run(
+          `
               MATCH (parent:Location {id: $parentId}), (child:Location {id: $childId})
               MERGE (child)<-[:IS_IN]-(parent)
               RETURN child.id, parent.id
             `,
-            {
-              parentId: parent.id,
-              childId: ctx.id,
-            },
-          )
-        })
-        parent = ctx
-      } finally {
-        session.close()
-      }
+          {
+            parentId: parent.id,
+            childId: ctx.id,
+          },
+        )
+      })
+      parent = ctx
     })
   }
   // delete all current locations from user and add new location
-  try {
-    await session.writeTransaction(transaction => {
-      return transaction.run(
-        `
+  await session.writeTransaction(transaction => {
+    return transaction.run(
+      `
           MATCH (user:User {id: $userId})-[relationship:IS_IN]->(location:Location)
           DETACH DELETE relationship
           WITH user
-          MATCH (location:Location {id: $locationId}) 
-          MERGE (user)-[:IS_IN]->(location) 
+          MATCH (location:Location {id: $locationId})
+          MERGE (user)-[:IS_IN]->(location)
           RETURN location.id, user.id
         `,
-        { userId: userId, locationId: data.id },
-      )
-    })
-  } finally {
-    session.close()
-  }
```


why are you no longer closing the session here in this function?
is it closed later?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-12-19T14:45:22Z" title="Thursday, December 19th 2019, 3:45:22 pm +01:00">Dec 19, 2019</time>_

---


See above :point_up: